provisioning/device/samples/solutions/BestPracticeSampleX509/readme.md

```diff
@@ -0,0 +1,139 @@
+
+[X509DpsSetup.ps1](https://github.com/Azure/azure-iot-sdk-csharp/tree/previews/v2/provisioning/device/samples/solutions/BestPracticeSampleX509/X509DpsSetup.ps1) creates an enrollment group in your DPS instance using the generated intermediate certificate.
+You can specify initial twin state, provisioning status, device capabilities, IoT hub name, eTag, etc. To learn more about these optional parameters, see [here](https://learn.microsoft.com/cli/azure/iot/dps/enrollment-group?view=azure-cli-latest#az-iot-dps-enrollment-group-create).
+
+### Provision a device through DPS and connect to IoT Hub
+
+In this step, we will use the chained device certificate to provision a device to an IoT Hub using the enrollment group. Devices provisioned through the same enrollment group will share the same initial configuration and will be assigned to one of the linked IoT Hub(s).
+
+1. Obtain the IDScope of the DPS instance from Azure Portal.
+2. From terminal, navigate to the [X509 sample folder](</provisioning/device/samples/getting started/X509Sample>) (where the .csproj file lives). Then execute the following command and check for build errors:
+
+```powershell
+    dotnet build
+```
+
+3. Set the IdScope, device certificate, and certificate password and run the sample. This will assign a device to an IoT hub and connect to the IoT hub and send a sample telemetry message. Check the log for the assigned IoT hub hostname.
+
+```powershell
+    dotnet run --s <IdScope> --n <Device certificate pfx file> --p <your password>
+```
+
+4. From terminal, navigate to the [device reconnection sample folder](</iothub/device/samples/how to guides/DeviceReconnectionSample>). Read [readme.md](</iothub/device/samples/how to guides/DeviceReconnectionSample/readme.md>) to learn how to initialize the device client, send device to cloud telemetry, receive cloud to device message, receive twin desired property update notifications, and update device twin's reported properties.
+
+    Open [DeviceReconnectionSample.cs](</iothub/device/samples/how to guides/DeviceReconnectionSample/DeviceReconnectionSample.cs>) in code editor. Find
+
+    ```csharp
+    s_deviceClient = new IotHubDeviceClient(_deviceConnectionStrings.First(), _clientOptions);
+    ```
+
+    and replace it with
+
+    ```csharp
+    var deviceCert = new X509Certificate2(_certificatePath, _certificatePassword);
+    var auth = new ClientAuthenticationWithX509Certificate(deviceCert, _deviceId);
+    s_deviceClient = new IotHubDeviceClient(_hostname, auth, _clientOptions);
+    ```
+
+    Then execute the following command and check for build errors:
+
+    ```powershell
+        dotnet build
+    ```
+
+5. Set the IoT hub hostname, device Id, device certificate password and path and run the sample.
+
+```powershell
+    dotnet run --h <assigned IoT hub hostname> --d <device Id> --p <your password> --n <path to device certificate pfx file>
+```
+
+![x509-bootsequence](media/bootsequence.png)
```

Comment:
auth flow and the boot sequence diagram seems to have different info. Let's double check these:
1) the password: is that send to the DPS or within the SDK?
2) do we need to separate the device process created by the users vs the SDK processes?
3) between #2 and #3, is there an additional process to establish connection with IoTHub?

Reply:
1. Certificate password is used within the SDK, not sent to DPS. I will make an update in the diagram.
2. X509Sample + DeviceReconnection samples combined is the device process. 
3. Nope.

Reply:
Agree on the #1 for the diagram update.
For #2 and #3 points, there are 2 different clients in the SDK(s) that we have used, maybe we just need to have 2 different box to indicate a DPS client and a IoTHub client?